backend/lambdas/amplify-deployer/index.py

```
@@ -0,0 +1,84 @@
+    response = s3.get_object(Bucket=source_bucket, Key=source_key)
+    zip_data = response["Body"].read()
+
+    # 2. Create Amplify deployment (get presigned upload URL)
+    logger.info(f"Creating deployment for app {app_id}, branch {branch_name}")
+    deployment = amplify.create_deployment(appId=app_id, branchName=branch_name)
+    upload_url = deployment["zipUploadUrl"]
+    job_id = deployment["jobId"]
+
+    # 3. Upload zip to presigned URL
+    logger.info(f"Uploading {len(zip_data)} bytes to Amplify")
+    req = urllib.request.Request(upload_url, data=zip_data, method="PUT")
```

Ruff (0.14.14)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)